physics/module_MYNNPBL_wrapper.F90

```diff
@@ -272,17 +282,25 @@ SUBROUTINE mynnedmf_wrapper_run(        &
       real(kind=kind_phys), dimension(im), intent(inout) ::              &
      &        pblh
       real(kind=kind_phys), dimension(im), intent(out) ::                &
-     &        ch,dtsfc1,dqsfc1,                                          &
+     &        ch,dtsfc1,dqsfc1,dusfc1,dvsfc1,                            &
      &        dtsfci_diag,dqsfci_diag,dtsfc_diag,dqsfc_diag,             &
+     &        dusfci_diag,dvsfci_diag,dusfc_diag,dvsfc_diag,             &
      &        maxMF
-     integer, dimension(im), intent(inout) ::                            &
-    &        kpbl,nupdraft,ktop_plume
+      integer, dimension(im), intent(inout) ::                           &
+     &        kpbl,nupdraft,ktop_plume
+
+      real(kind=kind_phys), dimension(im), intent(inout) ::              &
```

Comment:
Are these arrays always allocated? I believe not. If not, then you have to use (:) instead of (im). 

Also, I don't see "del" being used in the code (I may be missing them). If no, please don't pass it in.

Reply:
GitHub diff views are confusing. I was referring to the _cpl arrays. THese are only allocated if cpflx is true:
```
    if (Model%cplflx) then
      !--- incoming quantities
      allocate (Coupling%slimskin_cpl (IM))
      allocate (Coupling%dusfcin_cpl  (IM))
      allocate (Coupling%dvsfcin_cpl  (IM))
      allocate (Coupling%dtsfcin_cpl  (IM))
      allocate (Coupling%dqsfcin_cpl  (IM))
      allocate (Coupling%ulwsfcin_cpl (IM))
      allocate (Coupling%tseain_cpl   (IM))
      allocate (Coupling%tisfcin_cpl  (IM))
      allocate (Coupling%ficein_cpl   (IM))
      allocate (Coupling%hicein_cpl   (IM))
      allocate (Coupling%hsnoin_cpl   (IM))

      Coupling%slimskin_cpl = clear_val
      Coupling%dusfcin_cpl  = clear_val
      Coupling%dvsfcin_cpl  = clear_val
      Coupling%dtsfcin_cpl  = clear_val
      Coupling%dqsfcin_cpl  = clear_val
      Coupling%ulwsfcin_cpl = clear_val
      Coupling%tseain_cpl   = clear_val
      Coupling%tisfcin_cpl  = clear_val
      Coupling%ficein_cpl   = clear_val
      Coupling%hicein_cpl   = clear_val
      Coupling%hsnoin_cpl   = clear_val

      !--- accumulated quantities
      allocate (Coupling%dusfc_cpl  (IM))
      allocate (Coupling%dvsfc_cpl  (IM))
      allocate (Coupling%dtsfc_cpl  (IM))
      allocate (Coupling%dqsfc_cpl  (IM))
      allocate (Coupling%dlwsfc_cpl (IM))
      allocate (Coupling%dswsfc_cpl (IM))
      allocate (Coupling%dnirbm_cpl (IM))
      allocate (Coupling%dnirdf_cpl (IM))
      allocate (Coupling%dvisbm_cpl (IM))
      allocate (Coupling%dvisdf_cpl (IM))
      allocate (Coupling%nlwsfc_cpl (IM))
      allocate (Coupling%nswsfc_cpl (IM))
      allocate (Coupling%nnirbm_cpl (IM))
      allocate (Coupling%nnirdf_cpl (IM))
      allocate (Coupling%nvisbm_cpl (IM))
      allocate (Coupling%nvisdf_cpl (IM))

      Coupling%dusfc_cpl  = clear_val
      Coupling%dvsfc_cpl  = clear_val
      Coupling%dtsfc_cpl  = clear_val
      Coupling%dqsfc_cpl  = clear_val
      Coupling%dlwsfc_cpl = clear_val
      Coupling%dswsfc_cpl = clear_val
      Coupling%dnirbm_cpl = clear_val
      Coupling%dnirdf_cpl = clear_val
      Coupling%dvisbm_cpl = clear_val
      Coupling%dvisdf_cpl = clear_val
      Coupling%nlwsfc_cpl = clear_val
      Coupling%nswsfc_cpl = clear_val
      Coupling%nnirbm_cpl = clear_val
      Coupling%nnirdf_cpl = clear_val
      Coupling%nvisbm_cpl = clear_val
      Coupling%nvisdf_cpl = clear_val

      !--- instantaneous quantities
      allocate (Coupling%dusfci_cpl  (IM))
      allocate (Coupling%dvsfci_cpl  (IM))
      allocate (Coupling%dtsfci_cpl  (IM))
      allocate (Coupling%dqsfci_cpl  (IM))
      allocate (Coupling%dlwsfci_cpl (IM))
      allocate (Coupling%dswsfci_cpl (IM))
      allocate (Coupling%dnirbmi_cpl (IM))
      allocate (Coupling%dnirdfi_cpl (IM))
      allocate (Coupling%dvisbmi_cpl (IM))
      allocate (Coupling%dvisdfi_cpl (IM))
      allocate (Coupling%nlwsfci_cpl (IM))
      allocate (Coupling%nswsfci_cpl (IM))
      allocate (Coupling%nnirbmi_cpl (IM))
      allocate (Coupling%nnirdfi_cpl (IM))
      allocate (Coupling%nvisbmi_cpl (IM))
      allocate (Coupling%nvisdfi_cpl (IM))
      allocate (Coupling%t2mi_cpl    (IM))
      allocate (Coupling%q2mi_cpl    (IM))
      allocate (Coupling%tsfci_cpl   (IM))
      allocate (Coupling%psurfi_cpl  (IM))
      allocate (Coupling%oro_cpl     (IM))
      allocate (Coupling%slmsk_cpl   (IM))

      Coupling%dusfci_cpl  = clear_val
      Coupling%dvsfci_cpl  = clear_val
      Coupling%dtsfci_cpl  = clear_val
      Coupling%dqsfci_cpl  = clear_val
      Coupling%dlwsfci_cpl = clear_val
      Coupling%dswsfci_cpl = clear_val
      Coupling%dnirbmi_cpl = clear_val
      Coupling%dnirdfi_cpl = clear_val
      Coupling%dvisbmi_cpl = clear_val
      Coupling%dvisdfi_cpl = clear_val
      Coupling%nlwsfci_cpl = clear_val
      Coupling%nswsfci_cpl = clear_val
      Coupling%nnirbmi_cpl = clear_val
      Coupling%nnirdfi_cpl = clear_val
      Coupling%nvisbmi_cpl = clear_val
      Coupling%nvisdfi_cpl = clear_val
      Coupling%t2mi_cpl    = clear_val
      Coupling%q2mi_cpl    = clear_val
      Coupling%tsfci_cpl   = clear_val
      Coupling%psurfi_cpl  = clear_val
      Coupling%oro_cpl     = clear_val  !< pointer to sfcprop%oro
      Coupling%slmsk_cpl   = clear_val  !< pointer to sfcprop%slmsk
    endif
```